### ISSUE TITLE
Specially denote system fields in `diff`

### ISSUE THREAD

**Issue by hausdorff** [ksonnet/kubecfg#71]
Running `diff` often results in a bunch of system fields being reported, like this:

<img width="879" alt="screen shot 2017-08-01 at 10 48 36 am" src="https://user-images.githubusercontent.com/1409156/28839006-15d4af1e-76a7-11e7-91c9-7bcd382ced9f.png">

This is useful sometimes, but often you want to be able to quickly assess the changes to the fields the user has control out of. It seems like it might be sensible to default to not reporting system fields, and also have a flag like `--include-system-fields` or something to specifically opt into the behavior.

**Comment by anguslees** [comment]
Agreed, the default diff output is pretty basic.  Try `--diff-strategy=subset` and see if that works better for your use case.

The default diff is really only as terrible as it is because I'm lazy and have done only the most basic "fetch both objects and pump through a diff library".  We should definitely make an attempt to clean it up by at least using `export=true` (as soon as I can work out how to use `ExportOptions` through the go-client API).

**Comment by seh** [comment]
Be careful to note that when requesting a resource set (e.g., _kubectl get resourcequotas --export=true_), only the top-level _v1.List_ gets the "export" treatment, and none of the objects in the list get it. Apparently the API server doesn't treat the "export" option transitively.

**Comment by anguslees** [comment]
@seh, yes having tried export=true briefly, I'm a bit unimpressed.  I think we'll have to strip "known server-side" metadata in the client (even _with_ export=true).

**Comment by seh** [comment]
[Here's the recent thread](https://kubernetes.slack.com/archives/C09NXKJKA/p1505501820000341) from the "kubernetes-users" channel in the "Kubernetes" Slack team where I fumbled my way to this recognition. There are a few alternate approaches to cleaning unwanted fields linked from there.
